plugins/scaffolder/api-report.md

```diff
@@ -132,6 +132,18 @@ export type FieldExtensionOptions<
   validation?: CustomFieldValidator<TFieldReturnValue>;
 };
 
+// @public
+export type LastStepFormProps = {
+  disableButtons: boolean;
+  finishButtonLabel?: string;
+  formData: Record<string, any>;
+  handleBack: () => void;
+  handleCreate: () => void;
+  handleReset: () => void;
+  onFinish?: () => Promise<void>;
+  steps: Step[];
```

Comment:
Do we need steps here?

Reply:
Definitely. It's to preserve the same input to the original component and being able to construct the review step differently. And the most interesting data comes from steps 😄 

<img width="1333" alt="Screenshot 2022-10-24 at 20 46 41" src="https://user-images.githubusercontent.com/2265094/197602382-bf65467e-6f9b-4a07-a8f7-5c0fa2e7884f.png">

Reply:
Ah OK, I wonder if it makes sense to have a look at the props for the `ReviewState` component in the `next` folder as this is going to be the new component that's responsible for rendering this view. Ideally don't want to add one interface to break it straight away when we move to next, so maybe it makes sense to take on the burden now.

Do you think you could implement the `ReviewStepElement` props with something like this as the interface?

```ts
export interface ReviewStepProps {
  formData: JsonObject;
  handleBack: () => void;
  handleReset: () => void;
  handleCreate: () => void;
  steps: {
    uiSchema: UiSchema;
    mergedSchema: JsonObject;
    schema: JsonObject;
  }[];
}
```

I think that these props are the minimal that we can use for a default implementation, and we can provide another implementation of the `ReviewStep` for the `dryRun` editor instead rather than having these options baked in.

It does mean that we might have to update the `getReviewData` method, as it needs to take in that new `step` format. But you can also look at the new implementation how that works under the hood in the `next` folder instead too, with the new `json-schema-library` instead and copy that implementation for now.

Sorry for the backtracking a little bit, but I just realised that those props are only there because of other usage inside the Backstage project, and they're not used externally.

Reply:
Oh ok, i'll have a look at this direction 👍